DuckDuckGo/AppDelegate.swift

```diff
@@ -354,6 +286,18 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return true
     }
 
+    private func assertBookmarksMigrationHappensOnlyOnce() {
+        let key = UserDefaultsWrapper<Any>.Key.bookmarksLastGoodVersion.rawValue
+        if let lastGoodVersion = UserDefaults.standard.string(forKey: key) {
+            Pixel.fire(pixel: .debugBookmarksLost, withAdditionalParameters: [
+                PixelParameters.bookmarksLastGoodVersion: lastGoodVersion
+            ])
+            assertionFailure("Unexpected bookmarks migration")
```

Comment:
When migrating from a pre-FFS-favorites version, I'm getting this assertion on every app launch. This is because [this migration code](https://github.com/duckduckgo/BrowserServicesKit/blob/main/Sources/Bookmarks/Migrations/BookmarkFormFactorFavoritesMigration.swift#L65-L78) is apparently being run every time with a non-nil `orderedUUIDs` array. This means we have to fix the migration code in BSK, before we merge this PR, and we probably want to do it quick (e.g. on Monday). I'll create a task to track it.